implementations/code_df/new_contributors_of_commits.py

```diff
@@ -0,0 +1,139 @@
+import pandas as pd
+from datetime import datetime
+
+import utils
+import conditions
+from commit import Commit
+
+
+class NewContributorsOfCommits(Commit):
+    """
+    Class for New contributors of commits.
+    """
+
+    def compute(self, check_range=(None, None)):
+        """
+        Count number of new committers who committed between the two dates
+        of check range.
+
+        :param check_range: A tuple which represents the start and end date
+            when new committers will be considered
+
+        :returns count: the number of new committers who committed
+            between the dates of check_range
+        """
+
+        self.check_since, self.check_until = check_range
+
+        df = pd.DataFrame(columns=self.df.columns)
+
+        if self.check_since is not None:
+            df = self.df[self.df['created_date'] < self.check_since]
+
+        committers_before_check_since = set(df['author'].tolist())
+
+        df_new_committers = self.df['author'][
+                    (~self.df['author']
+                        .isin(
+                            list(committers_before_check_since)))
+                ]
+
+        if self.check_since is not None:
+            df_new_committers = df_new_committers[
+                self.check_since <= self.df['created_date']]
+
+        if self.check_until is not None:
+            df_new_committers = df_new_committers[
+                self.check_until >= self.df['created_date']]
+
+        count_new_committers = len(df_new_committers.unique())
+        return count_new_committers
+
+    def _agg(self, df, period):
+        """
+        Call an aggregation operation on a DataFrame or Series
+        to count the number of new committers in a period when
+        compared to the previous period.
+
+        :param df: a pandas DataFrame on which the aggregation will be
+            applied.
+
+        :param period: A string which can be any one of the pandas time
+            series rules:
+            'W': week
+            'M': month
+            'D': day
+
+        :returns df: The final aggregated DataFrame
+        """
+
+        def __new_committers_count(series):
```

Comment:
@jgbarah, according to our current implementation of `time_series()` and `_agg`, we would required implementing a separate aggregation method since we can't use the standard `count`, `mean`, etc. In this case, it is `__new_committers_count`. 

Would you prefer it being nested inside `_agg`  or as a separate but private/protected method?
Please let me know. The current implementation is, of course, a nested one.